client/landing/stepper/utils/path.ts

```diff
@@ -41,3 +43,29 @@ export function useLangRouteParam() {
 	const match = useMatch( path );
 	return match?.params.lang;
 }
+
+export const useLoginUrl = ( params: {
+	flowName?: string;
+	redirectTo?: string;
+	pageTitle?: string;
+} ): string => {
+	const locale = useLocale();
+
+	const loginPath =
+		locale && locale !== 'en' ? `/start/account/user/${ locale }` : `/start/account/user`;
+
+	const nonEmptyQueryParameters = Object.entries( params )
+		.filter( ( [ , value ] ) => value )
+		.map( ( [ key, value ] ) => {
+			switch ( key ) {
+				case 'redirectTo':
+					return [ 'redirect_to', value ];
+				case 'flowName':
+					return [ 'variationName', value ];
+				default:
+					return [ key, value ];
+			}
+		} );
```

Comment:
Did we need all of this complexity? It's not even clear to me what it's trying to do and whether even necessary

Reply:
This code tries to 
1 - filter - ignore parameters that are empty
2 - map transforms existing variable names to query params for the login escape path

And then generates a brand new object from it to feed to the `addQueryArgs` function.

Happy to go with a simpler method 👍 . Is there any particular approach you have in mind?

Reply:
Spread them?

Reply:
Okay now I'm curious too. I'm preparing to 🤯 from Christos's secret magic

Reply:
This logic looks very primitive, pre-ES6 stuff. I wonder if we have these things laying around in Calypso somewhere. Is it something that can't be done with simple conditional spreading?

```
{
    ...( flowName && { variationName: flowName } ),
    ...( redirectTo && { redirect_to: redirectTo } ),
    ...( pageTitle && { pageTitle } ),
}
```

Reply:
> Is it something that can't be done with simple conditional spreading?

I haven't looked much at the code beyond the signature, so this would be my immediate "reach" for something like this. that we resolve to logic that pertains to `convert to array -> filter -> map -> switch -> convert back to object` ought to be quite intentional, so probably missing something in the rest of it 🤔

Reply:
Thanks @chriskmnds that looks definitely much better and cleaner than my approach 🙌 . The `...( flowName && ` part did not immediately resonate in my brain but is a fantastic use of spread.

Reply:
Very cool thanks for sharing @chriskmnds 🙏

Reply:
Can you submit a PR to update this, please? It's a superfluous and imperative pattern that shouldn't be setting a precedent. :/